### PR TITLE
Extend & improve file operations tests

### PR DESCRIPTION
**Pull request by k-boikov**
### Background
Try to improve code coverage.
Switch from Unittest to Pytest

### Changes
Migrate tests for file operations

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by roeiba — DISMISSED** [review]

File: tests/unit/test_file_operations.py
Comment (Pwuts) [inline tests/unit/test_file_operations.py:1]: Please refactor to `pytest` style, and use available `workspace` or `workspace_root` fixtures

**Review by Pwuts — CHANGES_REQUESTED** [review]

File: tests/unit/test_file_operations.py
Comment (k-boikov) [reply tests/unit/test_file_operations.py:1]: Done.

File: tests/conftest.py
Comment (Pwuts) [inline tests/conftest.py:1]: This is unnecessary abstraction; path can be retrieved from `config.file_logger_path`

File: tests/test_file_operations.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    assert "Got an HTTP Error whilst trying to download file" in download_file(
+        url, local_name
+    )
+
+    url = "https://thiswebsiteiswrong.hmm/v0.0.0.tar.gz"
+    assert "Failed to establish a new connection:" in download_file(url, local_name)
+
+
+if __name__ == "__main__":
+    pytest.main()
```
Comment (Pwuts) [inline tests/test_file_operations.py]: This is not necessary

File: tests/test_file_operations.py
Comment (Pwuts) [inline tests/test_file_operations.py:1]: Why the move? This makes it a lot harder to see what changed.

**Review by Pwuts — CHANGES_REQUESTED** [review]

File: tests/conftest.py
Comment (k-boikov) [reply tests/conftest.py:1]: True, fixed.

File: tests/test_file_operations.py (hunk 1 above)
Comment (k-boikov) [reply tests/test_file_operations.py]: Took if from another test file. :angel: Removed.

File: tests/test_file_operations.py
Comment (k-boikov) [reply tests/test_file_operations.py:1]: Thought Unit folder is for unit tests only. Moved it back.

File: tests/unit/test_file_operations.py [hunk 2]
```diff
@@ -1,131 +1,169 @@
...
+
+    # Case 2: Search for a file that does not exist and make sure we don't throw
+    non_existent_file = "non_existent_file.txt"
+    files = search_files("")
+    assert non_existent_file not in files
+
+
+def test_download_file():
+    url = "https://github.com/Significant-Gravitas/Auto-GPT/archive/refs/tags/v0.2.2.tar.gz"
+    local_name = os.path.join(gettempdir(), "auto-gpt.tar.gz")
```
Comment (Pwuts) [inline tests/unit/test_file_operations.py:153]: there is a fixture you can use instead of `gettempdir()`, see conftest.py

**Review by Pwuts — APPROVED** [review]

**Pull request merged**